Nightly failing on `pytest keras/applications`

After the change to combine `actions.yml` and `nightly.yml` (#18757), the nightly job failing on `keras/applicaitons` test.  These tests are skipped in PR merges and were not included previously in `nightly.yml`.  We could revert the PR, but wanted to see if we can fix this issue.

Error Log: https://github.com/keras-team/keras/actions/runs/6838238049


Torch - 
```
Run pytest keras/applications --cov=keras.applications
  pytest keras/applications --cov=keras.applications
  coverage xml --include='keras/applications/*' -o apps-coverage.xml
  shell: /usr/bin/bash -e {0}
  env:
    PYTHON: 3.9
    KERAS_BACKEND: torch
    pythonLocation: /opt/hostedtoolcache/Python/3.9.18/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.9.18/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.18/x64
    Python[2](https://github.com/keras-team/keras/actions/runs/6838238049/job/18595121037#step:8:2)_ROOT_DIR: /opt/hostedtoolcache/Python/[3](https://github.com/keras-team/keras/actions/runs/6838238049/job/18595121037#step:8:3).9.18/x6[4](https://github.com/keras-team/keras/actions/runs/6838238049/job/18595121037#step:8:4)
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.9.18/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.9.18/x64/lib
ImportError while loading conftest '/home/runner/work/keras/keras/conftest.py'.
conftest.py:11: in <module>
    import torch  # noqa: F401
/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/torch/__init__.py:1119: in <module>
    from ._tensor import Tensor
/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/torch/_tensor.py:21: in <module>
    from torch.overrides import (
/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/torch/overrides.py:1[5](https://github.com/keras-team/keras/actions/runs/6838238049/job/18595121037#step:8:5)[9](https://github.com/keras-team/keras/actions/runs/6838238049/job/18595121037#step:8:9)1: in <module>
    has_torch_function = _add_docstr(
E   RuntimeError: function '_has_torch_function' already has a docstring
Error: Process completed with exit code 4.
```

--- a/.github/workflows/actions.yml
+++ b/.github/workflows/actions.yml
@@ -4,7 +4,6 @@ on:
   push:
     branches: [ master ]
   pull_request:
-  workflow_call:
   release:
     types: [created]
 

--- a/.github/workflows/nightly.yml
+++ b/.github/workflows/nightly.yml
@@ -9,18 +9,23 @@ permissions:
   contents: read
 
 jobs:
-  run-test-for-nightly:
-    uses: ./.github/workflows/actions.yml
-  nightly:
-    name: Build Wheel file and upload
-    needs: [run-test-for-nightly]
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.9]
+        backend: [tensorflow, jax, torch, numpy]
+    name: Run tests
     runs-on: ubuntu-latest
+    env:
+      PYTHON: ${{ matrix.python-version }}
+      KERAS_BACKEND: ${{ matrix.backend }}
     steps:
       - uses: actions/checkout@v3
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
-          python-version: 3.9
+          python-version: ${{ matrix.python-version }}
       - name: Get pip cache dir
         id: pip-cache
         run: |
@@ -31,6 +36,64 @@ jobs:
         with:
           path: ${{ steps.pip-cache.outputs.dir }}
           key: ${{ runner.os }}-pip-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements.txt') }}
+      - name: Install dependencies
+        run: |
+          pip install -r requirements.txt --progress-bar off --upgrade
+          pip uninstall -y keras keras-nightly
+          pip install -e "." --progress-bar off --upgrade
+      - name: Test integrations
+        if: ${{ matrix.backend != 'numpy'}}
+        run: |
+          python integration_tests/import_test.py
+      - name: Test TF-specific integrations
+        if: ${{ matrix.backend == 'tensorflow'}}
+        run: |
+          python integration_tests/tf_distribute_training_test.py
+      - name: Test Torch-specific integrations
+        if: ${{ matrix.backend == 'torch'}}
+        run: |
+          pytest integration_tests/torch_workflow_test.py
+      - name: Test with pytest
+        run: |
+          pytest keras --ignore keras/applications --cov=keras
+
+  format:
+    name: Check the code format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+      - name: Get pip cache dir
+        id: pip-cache
+        run: |
+          python -m pip install --upgrade pip setuptools
+          echo "dir=$(pip cache dir)" >> $GITHUB_OUTPUT
+      - name: pip cache
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.pip-cache.outputs.dir }}
+          key: ${{ runner.os }}-pip-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements.txt') }}
+      - name: Install dependencies
+        run: |
+          pip install -r requirements.txt --progress-bar off --upgrade
+          pip uninstall -y keras keras-nightly
+          pip install -e "." --progress-bar off --upgrade
+      - name: Lint
+        run: bash shell/lint.sh
+
+  nightly:
+    name: Build Wheel file and upload
+    needs: [build, format]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip setuptools

